getversion

```diff
@@ -1,46 +1,119 @@
 #!/usr/bin/env bash
+# ======================================================================
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# -----------------------------------------------------------------------------
+# getversion — determine the project version for packaging and diagnostics
+# -----------------------------------------------------------------------------
+#
+# This script computes a version string using the following logic:
+#
+#   1. Extract the base PACKAGE_VERSION from the `configure` script.
+#
+#   2. If running inside a Git repository and `git describe` is usable:
+#
+#      a. If the current commit matches an annotated tag exactly:
+#           VERSION = <tag>
+#         e.g.,     VERSION = 2.0.0+RC1
+#
+#      b. If the current commit is ahead of a tag:
+#           VERSION = <PACKAGE_VERSION>+dev.<N>.g<commit>
+#         e.g.,     VERSION = 2.0.1devel+dev.3.gabcd123
+#
+#   3. If *not* in a Git repo but a `VERSION` file exists:
+#         VERSION and BUILDNUMBER are read from that file (format: "<version> build <build_number>")
+#
+#   4. If neither a Git repo nor a VERSION file is present:
+#         VERSION = <PACKAGE_VERSION> from configure
+#         BUILDNUMBER = dev
+#
+#   5. If a BUILD_NUMBER file is present and BUILDNUMBER is "dev", it will override the default:
+#         BUILDNUMBER = $(cat BUILD_NUMBER)
+#
+# Output:
+#   - Default: <version> build <build_number>
+#   - With --short: just <version>
+#
+# Examples:
+#   ./getversion
+#       → 2.0.1devel+dev.3.gabcd123 build dev
```

Comment:
modify related version to `2.0.1-devel`

Reply:
Good catch! Although this is just an example, I will update it later for more accuracy.